sound/soc/sof/debug.c

```diff
@@ -812,6 +914,13 @@ int snd_sof_dbg_init(struct snd_sof_dev *sdev)
 		return err;
 #endif
 
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_IPC_MSG_INJECTOR)
+	err = snd_sof_debugfs_msg_inject_item(sdev, "ipc_msg_inject", 0644,
+					      &msg_inject_fops);
+	if (err < 0)
+		return err;
+#endif
+
```

Comment:
I may be missing something: we are moving the IPC flood test to the new SOF client framework. Why not do the same for the IPC message injector?

And along the same lines. could the IPC flood test use crafted messages? Is there any merit/rationale in keeping the two test capabilities separate?

Reply:
The SOF clients are not there yet and this feature has been requested with high urgency.

I will be moving it to a client and it is going to be the `sof-client-ipc-test.c`, which is atm in the draft PR is the ipc flood test. My only  concern with it is that we support more than one ipc_flood instance, thus more than one device for the ipc-test.
I guess I can use the same method to create the message injection only for a single device as I do with the symlink for the flood support.

If there is a need later we can add crafted message support to the flood, but it is really not in a scope of neither PR.